### PR TITLE
Update terraform_backend_defaults documentation

### PR DESCRIPTION
**Pull request by alyssa-sm**
### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -71,6 +71,10 @@ under `vars` before using an example blueprint.
 
 ### (Optional) Setting up a remote terraform state
 
+There are two ways to specify terraform backends in HPC Toolkit, defaults and per-group.
```
Comment (tpdownes) [inline examples/README.md:74]: Make "terraform backends" a link to:

https://developer.hashicorp.com/terraform/language/settings/backends/configuration

File: examples/README.md [hunk 2]
```diff
@@ -71,6 +71,10 @@ under `vars` before using an example blueprint.
...
+
+The first way is using terraform_backend_defaults at the top-level.
```
Comment (tpdownes) [inline examples/README.md:76]: Make `terraform_backend_defaults` fixed width text.

File: examples/README.md [hunk 3]
```diff
@@ -87,6 +91,12 @@ terraform_backend_defaults:
     impersonate_service_account: <<SERVICE_ACCOUNT>>
 ```
 
+It should be noted that "gcs" is used in the example above, but the configuration accepts anything that the backend accepts.
```
Comment (tpdownes) [inline examples/README.md:94]: I'd replace this with something like "All Terraform-supported backends are supported by the Toolkit." Specify the backend using `type` and its [configuration block](https://developer.hashicorp.com/terraform/language/settings/backends/configuration#using-a-backend-block) using `configuration`. 

For the [gcs](https://developer.hashicorp.com/terraform/language/settings/backends/gcs) backend, you must minimally supply the `bucket` configuration setting. The `prefix` setting is generated automatically as "blueprint_name/deployment_name/group_name" for each deployment group. This ensures uniqueness.

If you wish to specify a custom prefix, use a unique value for each group following this example:

```yaml
deployment_groups:
- group: example_group
  terraform_backend:
    type: gcs
    configuration:
      bucket: your-bucket
      prefix: your/object/prefix
```

**Review by tpdownes — CHANGES_REQUESTED** [review]
We need to link a bit more to Hashicorp docs and also give explicit example of a per-group setting.

File: examples/README.md (hunk 1 above)
Comment (alyssa-sm) [reply examples/README.md]: Done

File: examples/README.md (hunk 2 above)
Comment (alyssa-sm) [reply examples/README.md]: Done

File: examples/README.md (hunk 3 above)
Comment (alyssa-sm) [reply examples/README.md]: I think you phrased it nicely here so I replaced the section with your suggestion.

File: examples/README.md [hunk 4]
```diff
@@ -71,6 +71,10 @@ under `vars` before using an example blueprint.
 
 ### (Optional) Setting up a remote terraform state
 
+There are two ways to specify [terraform backends] in HPC Toolkit, defaults and per-group.
```
Comment (tpdownes) [inline examples/README.md:74]: > There are two ways to specify [terraform backends] in HPC Toolkit, defaults and per-group.

Let's say something more like

> There are two ways to specify [terraform backends] in HPC Toolkit: a default setting that propagates all groups and custom per-group configuration:
> 
> `terraform_backend_defaults` at top level of YAML blueprint
> `terraform_backend` within a deployment group definition
>
> Examples of each are shown below. If both settings are used, then the custom per-group value is used without modification.

**Review by tpdownes — CHANGES_REQUESTED** [review]
We're close, but I notice that we don't have "terraform_backend" mentioned in the prose, only in the YAML example.

File: examples/README.md (hunk 4 above)
Comment (alyssa-sm) [reply examples/README.md]: Done

**Review by tpdownes — CHANGES_REQUESTED** [review]

File: examples/README.md [hunk 5]
```diff
@@ -71,6 +71,13 @@ under `vars` before using an example blueprint.
 
 ### (Optional) Setting up a remote terraform state
 
+There are two ways to specify [terraform backends] in HPC Toolkit: a default setting that propagates all groups and custom per-group configuration:
+
+`terraform_backend_defaults` at top-level of YAML blueprint
```
[the remark this answers is not in the document]
Comment (alyssa-sm) [reply examples/README.md]: Good catch!

**Review by tpdownes — APPROVED** [review]

**Pull request merged**